### PR TITLE
Enable FX graph cache for huggingface and timm benchmarks

### PR DESCRIPTION
**Pull request by masnesral**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #126205

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

### CODE REVIEW

**Comment by masnesral** [comment]
Looks like it we actually get a cold-start improvement for huggingface: https://hud.pytorch.org/benchmark/compilers?startTime=Wed%2C%2008%20May%202024%2014%3A16%3A52%20GMT&stopTime=Wed%2C%2015%20May%202024%2014%3A16%3A52%20GMT&granularity=hour&suite=torchbench&mode=training&dtype=amp&lBranch=gh/masnesral/58/head&lCommit=ed23d4e6a6740c27a9174b8260ef58e47913f6fa&rBranch=main&rCommit=afda6685ae87cce7ac2fe4bac3926572da2960f7

Is that expected? Is there something about some of the huggingface models that we re-compile the same kernel more than once within one run? Maybe @Chillee has some insight.

**Comment by Chillee** [comment]
Interesting. Do we support re-sharing compiled graphs within the same run?

Oh also, if we don't support that I think it's worth supporting it.

**Comment by masnesral** [comment]
> Interesting. Do we support re-sharing compiled graphs within the same run?

Well the cache implementation supports it if that's what you're asking. Any time we're about to compile a graph, we consult the cache. So if an entry has been populated previously in the same process (or a prior process), then the cache will serve it. I guess I'm surprised to see that it's so prevalent in huggingface that we'd be compiling the same graph/inputs twice in the same process.

**Comment by Chillee** [comment]
> I guess I'm surprised to see that it's so prevalent in huggingface that we'd be compiling the same graph/inputs twice in the same process.

This actually isn't super shocking in this case to me. Transformers have a lot of identical blocks, and I know we have some amount of graph breaks in HF transformers.

**Review by eellison — APPROVED** [review]

File: benchmarks/dynamo/huggingface.py [hunk 1]
```diff
@@ -15,6 +15,9 @@
 
 log = logging.getLogger(__name__)
 
+# Enable FX graph caching
+torch._inductor.config.fx_graph_cache = True
```
Comment (eellison) [inline benchmarks/dynamo/huggingface.py:19]: Don't set if `TORCHINDUCTOR_FX_GRAPH_CACHE` is in the env ?

**Comment by masnesral** [comment]
None of the failures have anything to do with timm or huggingface benchmarks. They're either in torchbench (hf_BigBird) or in functorch unit tests.

**Pull request closed without merging**